app/components/navigation/google-login-btn.tsx

```diff
@@ -0,0 +1,55 @@
+import Script from "next/script";
+import { useEffect, useState, useRef } from "react";
+
+interface GoogleLogin {
+    credential: string,
+    select_by: string
+}
+
+export default function GoogleLoginButton(){
+    const [scriptLoaded, setScriptLoaded] = useState(false);
+    const handleScriptLoad = () => {
+        setScriptLoaded(true);
+    };
+
+    const googleLoginRef = useRef(null);
+
+    useEffect(()=> {
+        if(scriptLoaded){
```

Comment:
Dumb question: if we already have `scriptLoaded` as a dep, then how come we should check here to make sure it actually has?

Reply:
We only want to run this code if the script is loaded. Otherwise, it will error out

Reply:
Oh... I didn't fully read [the docs](https://react.dev/reference/react/useEffect#reference) on this one so that is my bad 🤦